Makefile

```diff
@@ -60,6 +60,18 @@ header:
 	@echo "----------------------------"
 	@echo
 
+cideps:
+	# Assumption: User of this file is a Super user
+	# Assumption: `apt` is present in system
+	# Assumption: `python` is present in system
+	apt install python-pip
+	pip install --upgrade pip
+	pip install pyYAML
+	pip install kubernetes
+
+ci: cideps
+	python test.py
```

Comment:
Please make use of linting tools to lint this code

Reply:
I have used `pylint` while writing Python code. Do you want me to lint Makefile?

Reply:
I would like the pylint as a make target. It should be run in the CI itself.

Reply:
OK, got it.

Reply:
I ran pylint in terminal and I found some Convension and some Warning. Can we ignore them or should we run pylint strictly?

Reply:
I shall push new code with less lint error.

Reply:
New code has lint errors less upto `Warning level`. (Specifically `Warning level` and `Convention level issues` only).

Reply:
abhishek@genesis:~/GoWorkspace/src/github.com/openebs/node-disk-manager$ pylint ndmtestpackage.py 
No config file found, using default configuration
************* Module ndmtestpackage
W:349, 0: Anomalous backslash in string: '\['. String constant might be missing an r prefix. (anomalous-backslash-in-string)
W:349, 0: Anomalous backslash in string: '\]'. String constant might be missing an r prefix. (anomalous-backslash-in-string)
C:  1, 0: Missing module docstring (missing-docstring)
C: 48, 0: Missing class docstring (missing-docstring)
C: 53, 0: Missing function docstring (missing-docstring)
C: 59, 0: Missing function docstring (missing-docstring)
C: 61, 4: Variable name "v1" doesn't conform to snake_case naming style (invalid-name)
C: 66, 4: Variable name "ndmPod" doesn't conform to snake_case naming style (invalid-name)
W: 81, 4: Else clause on loop without a break statement (useless-else-on-loop)
C: 67,10: Comparison to None should be 'expr is None' (singleton-comparison)
C: 78,16: Variable name "ndmPod" doesn't conform to snake_case naming style (invalid-name)
C: 82,11: Comparison to None should be 'expr is None' (singleton-comparison)
C: 88, 0: Missing function docstring (missing-docstring)
C: 91, 0: Missing function docstring (missing-docstring)
C:114, 0: Missing function docstring (missing-docstring)
C:125, 0: Missing function docstring (missing-docstring)
C:127,27: Variable name "f" doesn't conform to snake_case naming style (invalid-name)
C:128, 8: Variable name "ndmYaml" doesn't conform to snake_case naming style (invalid-name)
C:141,61: Variable name "f" doesn't conform to snake_case naming style (invalid-name)
C:144, 0: Missing function docstring (missing-docstring)
C:157, 0: Missing function docstring (missing-docstring)
W:165, 8: Statement seems to have no effect (pointless-statement)
C:188, 0: Missing function docstring (missing-docstring)
C:192, 8: Variable name "statusStr" doesn't conform to snake_case naming style (invalid-name)
C:203,11: Variable name "v" doesn't conform to snake_case naming style (invalid-name)
C:207, 0: Missing function docstring (missing-docstring)
W:210, 8: Unused variable 'status' (unused-variable)
C:219, 0: Missing function docstring (missing-docstring)
C:236,24: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
C:239, 0: Missing function docstring (missing-docstring)
C:241, 4: Variable name "v1" doesn't conform to snake_case naming style (invalid-name)
C:246, 0: Missing function docstring (missing-docstring)
C:248, 4: Variable name "v1" doesn't conform to snake_case naming style (invalid-name)
W:256, 4: Unused variable 'api_response' (unused-variable)
C:258, 0: Missing function docstring (missing-docstring)
C:266, 4: Variable name "c" doesn't conform to snake_case naming style (invalid-name)
C:276,27: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
W:285,15: Catching too general exception Exception (broad-except)
C:282,48: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
C:285,28: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
C:293,32: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
C:302, 0: Missing function docstring (missing-docstring)
C:304, 4: Variable name "v1" doesn't conform to snake_case naming style (invalid-name)
C:307, 0: Missing function docstring (missing-docstring)
C:322,13: Using type() instead of isinstance() for a typecheck. (unidiomatic-typecheck)
C:323,16: Using type() instead of isinstance() for a typecheck. (unidiomatic-typecheck)
C:326,19: Using type() instead of isinstance() for a typecheck. (unidiomatic-typecheck)
C:327,28: Using type() instead of isinstance() for a typecheck. (unidiomatic-typecheck)
C:328,23: Comparison to False should be 'not expr' or 'expr is False' (singleton-comparison)
C:344, 0: Missing function docstring (missing-docstring)
C:353, 0: Missing function docstring (missing-docstring)
W:369,11: Catching too general exception Exception (broad-except)
W:365,19: Catching too general exception Exception (broad-except)
W:377,11: Catching too general exception Exception (broad-except)
C:377,24: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
W: 15, 0: Unused import json (unused-import)
C: 11, 0: standard import "import subprocess" should be placed before "from kubernetes import client, config" (wrong-import-order)
C: 13, 0: standard import "from time import sleep" should be placed before "from kubernetes import client, config" (wrong-import-order)
C: 15, 0: standard import "import json" should be placed before "from kubernetes import client, config" (wrong-import-order)
C: 16, 0: standard import "from re import compile as regex_compile" should be placed before "from kubernetes import client, config" (wrong-import-order)
C: 18, 0: standard import "import os" should be placed before "from kubernetes import client, config" (wrong-import-order)
C: 19, 0: standard import "from os.path import isfile" should be placed before "from kubernetes import client, config" (wrong-import-order)
C: 20, 0: standard import "from os import remove as removefile" should be placed before "from kubernetes import client, config" (wrong-import-order)

------------------------------------------------------------------
Your code has been rated at 7.35/10 (previous run: 6.97/10, +0.38)

abhishek@genesis:~/GoWorkspace/src/github.com/openebs/node-disk-manager$ echo $?
20

Reply:
```
abhishek@genesis:~/GoWorkspace/src/github.com/openebs/node-disk-manager$ pylint test.py
No config file found, using default configuration
************* Module test
C:  1, 0: Missing module docstring (missing-docstring)
W:  6, 0: Wildcard import ndmtestpackage (wildcard-import)
W: 32,19: Catching too general exception Exception (broad-except)
C: 30,16: Constant name "status" doesn't conform to UPPER_CASE naming style (invalid-name)
C: 32,32: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
W: 58,15: Catching too general exception Exception (broad-except)
C: 58,28: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
C: 73,12: Constant name "ndmPod" doesn't conform to UPPER_CASE naming style (invalid-name)
C: 74,12: Constant name "ndmPodState" doesn't conform to UPPER_CASE naming style (invalid-name)
C: 93,31: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
C:101,28: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
C:108, 4: Constant name "ndmPod" doesn't conform to UPPER_CASE naming style (invalid-name)
C:114,16: Constant name "ndmLog" doesn't conform to UPPER_CASE naming style (invalid-name)
C:116,47: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
C:128,11: Comparison to True should be just 'expr' or 'expr is True' (singleton-comparison)
W:165,11: Catching too general exception Exception (broad-except)
C:143, 8: Constant name "lsblk_in_pod" doesn't conform to UPPER_CASE naming style (invalid-name)
C:148, 8: Constant name "lsblk_in_pod" doesn't conform to UPPER_CASE naming style (invalid-name)
W:162,15: Catching too general exception Exception (broad-except)
C:152,12: Constant name "lsblk_in_host" doesn't conform to UPPER_CASE naming style (invalid-name)
C:153,12: Constant name "lsblk_in_host" doesn't conform to UPPER_CASE naming style (invalid-name)
C:162,28: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
C:165,24: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
W:191,11: Catching too general exception Exception (broad-except)
C:173, 8: Constant name "ndm_in_pod" doesn't conform to UPPER_CASE naming style (invalid-name)
W:188,15: Catching too general exception Exception (broad-except)
C:176,12: Constant name "ndm_in_host" doesn't conform to UPPER_CASE naming style (invalid-name)
C:188,28: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
C:191,24: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
W:  6, 0: Unused import stream from wildcard import (unused-wildcard-import)
W:  6, 0: Unused import core_v1_api from wildcard import (unused-wildcard-import)
W:  6, 0: Unused import isfile from wildcard import (unused-wildcard-import)
W:  6, 0: Unused import NDMTestException from wildcard import (unused-wildcard-import)
W:  6, 0: Unused import get_node_name from wildcard import (unused-wildcard-import)
W:  6, 0: Unused import make from wildcard import (unused-wildcard-import)
W:  6, 0: Unused import yaml from wildcard import (unused-wildcard-import)
W:  6, 0: Unused import POD_BAD_STATES from wildcard import (unused-wildcard-import)
W:  6, 0: Unused import Configuration from wildcard import (unused-wildcard-import)
W:  6, 0: Unused import config from wildcard import (unused-wildcard-import)
W:  6, 0: Unused import POD_WAIT_PHASES from wildcard import (unused-wildcard-import)
W:  6, 0: Unused import DOCKER_IMAGE_TAG from wildcard import (unused-wildcard-import)
W:  6, 0: Unused import NDM_TEST_YAML_PATH from wildcard import (unused-wildcard-import)
W:  6, 0: Unused import remove_all_docker_containers from wildcard import (unused-wildcard-import)
W:  6, 0: Unused import NDM_YAML from wildcard import (unused-wildcard-import)
W:  6, 0: Unused import label_node from wildcard import (unused-wildcard-import)
W:  6, 0: Unused import err from wildcard import (unused-wildcard-import)
W:  6, 0: Unused import regex_compile from wildcard import (unused-wildcard-import)
W:  6, 0: Unused import removefile from wildcard import (unused-wildcard-import)
W:  6, 0: Unused import DOCKER_IMAGE_NAME from wildcard import (unused-wildcard-import)
W:  6, 0: Unused import clean from wildcard import (unused-wildcard-import)
W:  6, 0: Unused import os from wildcard import (unused-wildcard-import)
W:  6, 0: Unused import V1Pod from wildcard import (unused-wildcard-import)

------------------------------------------------------------------
Your code has been rated at 6.23/10 (previous run: 5.37/10, +0.86)

abhishek@genesis:~/GoWorkspace/src/github.com/openebs/node-disk-manager$ echo $?
20
```

Reply:
```
abhishek@genesis:~/GoWorkspace/src/github.com/openebs/node-disk-manager$ pylint clean.py 
No config file found, using default configuration
************* Module clean
C:  1, 0: Missing module docstring (missing-docstring)

------------------------------------------------------------------
Your code has been rated at 7.50/10 (previous run: 7.50/10, +0.00)

abhishek@genesis:~/GoWorkspace/src/github.com/openebs/node-disk-manager$ echo $?
16
```

Reply:
```
abhishek@genesis:~/GoWorkspace/src/github.com/openebs/node-disk-manager$ pylint k8s_adm.py
No config file found, using default configuration
************* Module k8s_adm
C:  1, 0: Missing module docstring (missing-docstring)
C: 28, 0: Missing function docstring (missing-docstring)
C: 56, 0: Missing function docstring (missing-docstring)
C: 59, 0: Missing function docstring (missing-docstring)
C: 82, 0: Missing function docstring (missing-docstring)
C: 95, 0: Missing function docstring (missing-docstring)
C:102, 0: Missing function docstring (missing-docstring)
C:114, 0: Missing function docstring (missing-docstring)
W:141,15: Catching too general exception Exception (broad-except)
C:158, 0: Missing function docstring (missing-docstring)

-------------------------------------------------------------------
Your code has been rated at 9.00/10 (previous run: 10.00/10, -1.00)

abhishek@genesis:~/GoWorkspace/src/github.com/openebs/node-disk-manager$ echo $?
20
```